### PR TITLE
Add support for GSSAPI authentication.

### PR DESCRIPTION
**Pull request by adelton**

### CODE REVIEW

**Comment by simo5** [comment]
can you add proper deps on request_gssapi and a test ?

File: src/custodia/cli/__init__.py [hunk 1]
```diff
@@ -133,6 +133,11 @@ def timeout(arg):
     help='PEM encoded key file (if not given, key is read from certfile)'
 )
 
+# Use Negotiate / GSSAPI
+main_parser.add_argument(
```
Comment (tiran) [inline src/custodia/cli/__init__.py]: Please make the option mutually exclusive with cert auth.

File: src/custodia/client.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 from jwcrypto.jwk import JWK
 
 import requests
+from requests_gssapi import HTTPSPNEGOAuth
```
Comment (tiran) [inline src/custodia/client.py]: I don't like the fact that all users have to install requests_gssapi. Please make the import optional and add an optional dependency to setup.py (``gssapi`` extras require).

**Comment by adelton** [comment]
Would you prefer deps, or maybe weakening the `import` and failing in runtime if the modules is not available?

As for tests, I'm not really sure how we would test it, given we'd likely need KDC for that ...

File: src/custodia/client.py [hunk 3]
```diff
@@ -84,6 +85,9 @@ def set_client_cert(self, certfile, keyfile=None):
         else:
             self.session.cert = (certfile, keyfile)
 
+    def set_gssapi_auth(self):
```
Comment (tiran) [inline src/custodia/client.py]: ``set_gssapi_auth`` should support passing arguments to ``HTTPSPNEGOAuth()``.

**Comment by frozencemetery** [comment]
Thanks for using requests_gssapi!  If tests are needed, it may be possible to use [k5test](https://github.com/pythongssapi/k5test).

That said, I'm not sure there's a lot of value in setting up a full test with KDC here, since bugs in that would be bugs in a lower component - perhaps it'd be better to check that it starts a GSSAPI handshake when prompted?

**Comment by tiran** [comment]
I don't think we need a full integration test or functional test here. A unit test is good enough, just check that the function sets the object.

**Comment by raildo** [comment]
Can you also rebase that patch? So, it would be also in a good shape passing the tests.

**Comment by adelton** [comment]
Closing in favour of https://github.com/latchset/custodia/pull/238.

**Pull request closed without merging**